src/main/java/org/kaazing/netx/ws/internal/io/WsInputStream.java

```diff
@@ -141,101 +145,106 @@ public int available() throws IOException {
 
     @Override
     public int read() throws IOException {
```

Comment:
Actually, we implement two read methods:

* public int read(byte[] buf, int offset, int length) throws IOException
* public int read() throws IOException

We acquire the lock in `read(byte[] buf, int offset, int length)` implementation.